grayscale.py

```diff
@@ -1,336 +1,1041 @@
+        self.bitmapSource2 = bitmapData2
+        self.bitmapByteCount = width*(height//8)
+        if(height%8):
+            self.bitmapByteCount+=width
+        self.frameCount = 1
+        self.currentFrame = 0
+        if type(self.bitmapSource1)==str:
+            self.bitmap = bytearray(self.bitmapByteCount1)
+            self.file1 = open(self.bitmapSource1,'rb')
+            self.file1.readinto(self.bitmap1)
+            self.frameCount = os.stat(self.bitmapSource1)[6] // self.bitmapByteCount
+        elif type(self.bitmapSource1)==bytearray:
+            self.bitmap1 = memoryview(self.bitmapSource1)[0:self.bitmapByteCount]
+            self.frameCount = len(self.bitmapSource1) // self.bitmapByteCount
+        if type(self.bitmapSource2)==str:
+            self.bitmap2 = bytearray(self.bitmapByteCount)
+            self.file2 = open(self.bitmapSource2,'rb')
+            self.file2.readinto(self.bitmap2)
+            assert(self.frameCount == os.stat(self.bitmapSource2)[6] // self.bitmapByteCount)
+        elif type(self.bitmapSource2)==bytearray:
+            self.bitmap2 = memoryview(self.bitmapSource2)[0:self.bitmapByteCount]
+            assert(self.frameCount == len(self.bitmapSource2) // self.bitmapByteCount)
+        self.x = x
+        self.y = y
+        self.key = key
+        self.mirrorX = mirrorX
+        self.mirrorY = mirrorY
+
+    @micropython.native
+    def getFrame(self):
+        return self.currentFrame
+
+    @micropython.native
+    def setFrame(self, frame):
+        if(frame >= 0 and (self.currentFrame is not frame % (self.frameCount))):
+            self.currentFrame = frame % (self.frameCount)
+            offset=self.bitmapByteCount*self.currentFrame
+            if type(self.bitmapSource1)==str:
+                self.file1.seek(offset)
+                self.file1.readinto(self.bitmap1)
+                #f.close()
+            elif type(self.bitmapSource1)==bytearray:
+                self.bitmap1 = memoryview(self.bitmapSource1)[offset:offset+self.bitmapByteCount]
+            if type(self.bitmapSource2)==str:
+                self.file2.seek(offset)
+                self.file2.readinto(self.bitmap2)
+                #f.close()
+            elif type(self.bitmapSource2)==bytearray:
+                self.bitmap2 = memoryview(self.bitmapSource2)[offset:offset+self.bitmapByteCount]
+
+
+# You can't use const() to define class properties, so we'll define them here
+
+# The times below are calculated using phase 1 and phase 2 pre-charge periods of
+# 1 clock.
+# Note that although the SSD1306 datasheet doesn't state it, the 50
+# clocks period per row _is_ a constant (datasheets for similar controllers from
+# the same manufacturer state this).
+# 530kHz is taken to be the highest nominal clock frequency. The calculations
+# shown provide the value in seconds, which can be multiplied by 1e6 to provide
+# a microsecond value.
+Grayscale_pre_frame_time_us    = const( 785)     # 8 rows: ( 8*(1+1+50)) / 530e3 seconds
```

Comment:
Underscore (_) prefixed module level consts should save more RAM, than without. Unless they should be accessible from outside?